server/engine/src/main/java/org/eclipse/lsp/cobol/implicitDialects/cics/utility/CICSDeleteOptionsCheckUtility.java

```diff
@@ -0,0 +1,121 @@
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *    Broadcom, Inc. - initial API and implementation
+ *
+ */
+
+package org.eclipse.lsp.cobol.implicitDialects.cics.utility;
+
+import org.antlr.v4.runtime.ParserRuleContext;
+import org.eclipse.lsp.cobol.common.dialects.DialectProcessingContext;
+import org.eclipse.lsp.cobol.common.error.ErrorSeverity;
+import org.eclipse.lsp.cobol.common.error.SyntaxError;
+import org.eclipse.lsp.cobol.implicitDialects.cics.CICSLexer;
+import org.eclipse.lsp.cobol.implicitDialects.cics.CICSParser;
+
+import java.util.*;
+
+import static org.eclipse.lsp.cobol.implicitDialects.cics.CICSParser.RULE_cics_delete;
+
+/** Checks CICS Delete rules for required and invalid options */
+public class CICSDeleteOptionsCheckUtility extends CICSOptionsCheckBaseUtility {
+
+  public static final int RULE_INDEX = RULE_cics_delete;
+
+  private static final Map<Integer, ErrorSeverity> DUPLICATE_CHECK_OPTIONS =
+          new HashMap<Integer, ErrorSeverity>() {
+            {
+              put(CICSLexer.DELETE, ErrorSeverity.ERROR);
+              put(CICSLexer.FILE, ErrorSeverity.ERROR);
+              put(CICSLexer.TOKEN, ErrorSeverity.ERROR);
+              put(CICSLexer.RIDFLD, ErrorSeverity.ERROR);
+              put(CICSLexer.KEYLENGTH, ErrorSeverity.ERROR);
+              put(CICSLexer.NUMREC, ErrorSeverity.ERROR);
+              put(CICSLexer.SYSID, ErrorSeverity.ERROR);
+              put(CICSLexer.ACTIVITY, ErrorSeverity.ERROR);
+              put(CICSLexer.CHANNEL, ErrorSeverity.ERROR);
+              put(CICSLexer.CONTAINER, ErrorSeverity.ERROR);
+              put(CICSLexer.COUNTER, ErrorSeverity.ERROR);
+              put(CICSLexer.DCOUNTER, ErrorSeverity.ERROR);
+              put(CICSLexer.POOL, ErrorSeverity.ERROR);
+              put(CICSLexer.EVENT, ErrorSeverity.ERROR);
+              put(CICSLexer.TIMER, ErrorSeverity.ERROR);
+              put(CICSLexer.GENERIC, ErrorSeverity.WARNING);
+              put(CICSLexer.NOSUSPEND, ErrorSeverity.WARNING);
+              put(CICSLexer.RBA, ErrorSeverity.WARNING);
+              put(CICSLexer.RRN, ErrorSeverity.WARNING);
+              put(CICSLexer.ACQACTIVITY, ErrorSeverity.WARNING);
+              put(CICSLexer.PROCESS, ErrorSeverity.WARNING);
+              put(CICSLexer.ACQPROCESS, ErrorSeverity.WARNING);
+            }
+          };
+
+  public CICSDeleteOptionsCheckUtility(
+      DialectProcessingContext context, List<SyntaxError> errors) {
+    super(context, errors, DUPLICATE_CHECK_OPTIONS);
+  }
+
+  /**
+   * Entrypoint to check CICS DELETE rule options
+   *
+   * @param ctx ParserRuleContext subclass containging options
+   * @param <E> A subclass of ParserRuleContext
+   */
+  public <E extends ParserRuleContext> void checkOptions(E ctx) {
+    if (ctx.getClass() == CICSParser.Cics_delete_group_oneContext.class) {
```

Comment:
See #2521 as a reference to what Chace is referring to. The idea is to use the rule index through a switch instead of bringing in reflection to find out what class we're looking at.